### PR TITLE
add 108 convert-sorted-array-to-binary-search-tree

### LINKED ISSUE #45: add 108 Convert Sorted Array to Binary Search Tree
## Summary
add 108

## Goal
add 108

## Todo
- [ ] add 108
- [ ] write time and space complexity

## Deadline
4/26

## Parent issue
None

## References
None

## Notes
None

### PR DESCRIPTION
**Pull request by kobakobashu**
time : 25m

### CODE REVIEW

**Comment by kobakobashu** [comment]
Please check my code.

File: 108.convert-sorted-array-to-binary-search-tree.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+class Solution:
+    def sortedArrayToBST(self, nums: List[int]) -> TreeNode:
+        """
+        time: O(N)  N : the total of all nodes
+        space: O(N)
+        """
+        if not nums:
+            return None
+        half = len(nums) // 2
+        node = TreeNode(nums[half], None, None)
```
Comment (ryuji0123) [inline 108.convert-sorted-array-to-binary-search-tree.py:23]: You don't need to specify values that are the same ones with default values.
```py
node = TreeNode(nums[half])
```

**Review by ryuji0123 — CHANGES_REQUESTED** [review]

**Comment by kobakobashu** [comment]
I fixed my code.  Please check it.

File: 108.convert-sorted-array-to-binary-search-tree.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+#         self.left = left
+#         self.right = right
+class Solution:
+    def sortedArrayToBST(self, nums: List[int]) -> TreeNode:
+        """
+        time: O(N)  N : the total of all nodes
+        space: O(N)
+        """
+        if not nums:
+            return None
```
Comment (ryuji0123) [inline 108.convert-sorted-array-to-binary-search-tree.py:21]: insert a line after return statement.

**Review by ryuji0123 — CHANGES_REQUESTED** [review]

**Comment by kobakobashu** [comment]
I fixed my code.  Please check it.

File: 108.convert-sorted-array-to-binary-search-tree.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+        half = len(nums) // 2
+        node = TreeNode(nums[half])
+
+        node.left = self.sortedArrayToBST(nums[:half])
+        node.right = self.sortedArrayToBST(nums[half+1:])
```
Comment (ryuji0123) [inline 108.convert-sorted-array-to-binary-search-tree.py:27]: Insert a line here.

**Review by ryuji0123 — CHANGES_REQUESTED** [review]
Check "Readable code"

**Pull request closed without merging**